jcstress-samples/src/main/java/org/openjdk/jcstress/samples/concurreny/mutex/Mutex_01_NoAlgorithm.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.jcstress.samples.concurreny.mutex;
+
+import org.openjdk.jcstress.annotations.Actor;
+import org.openjdk.jcstress.annotations.JCStressTest;
+import org.openjdk.jcstress.annotations.Outcome;
+import org.openjdk.jcstress.annotations.State;
+import org.openjdk.jcstress.infra.results.II_Result;
+import org.openjdk.jcstress.infra.results.ZZ_Result;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE;
+
+/*
+    How to run this test:
+        $ java -jar jcstress-samples/target/jcstress.jar -t Mutex_01_NoAlgorithm
+*/
+
+/**
+ * This sample demonstrates you how you can introduce a critical section to check algorithms
+ * which ensure only one actor at most can enter the critical section.
+ */
+@JCStressTest
+@Outcome(expect = ACCEPTABLE, desc = "Both actors have entered the critical section whenever they wanted")
+@State
+public class Mutex_01_NoAlgorithm {
```

Comment:
I think it would be great to add `SynchronizedAlgorithm` as well.
It would be the relevant middle-ground between `NoAltorithm` and `highly-sophisticated` Dekker/Peterson :)

Reply:
Yes, that's a good idea. I will do it.

Reply:
I added a sample with an atomic boolean. However, all examples use busy waiting so far.
Do you wish to add samples which get blocked instead of  busy waiting?

Reply:
Yes, do `synchronized` example too, please, @mmirwaldt-openjdk.

Reply:
In my opinion, half of the story here is to teach the readers, so if you write a trivial `synchronized(lock) { ... } `, then it would be way easier to understand what Dekker algorithm really means.

I think adding blocking primitives here is OK.

Reply:
Ok, I will

Reply:
I have now added two more samples: one for synchronized and one for ReentrantLock (ReentrantLock  just for completeness)